fiftyone/utils/torch.py

```diff
@@ -875,7 +876,7 @@ def _parse_output(self, output, frame_size, confidence_thresh):
                 fol.Keypoint(
                     label=self.class_labels[label],
                     points=points,
-                    confidence=score,
+                    visible=visible,
```

Comment:
I think you mentioned that models like `keypoint-rcnn-resnet50-fpn-coco-torch` expose keypoint confidences too? Can you parse and store those here?

Here's some example code you can use to test the model:

```py
import fiftyone as fo
import fiftyone.zoo as foz

dataset = foz.load_zoo_dataset(
    "coco-2017",
    split="validation",
    classes="person",
    max_samples=10,
    shuffle=True,
)

model = foz.load_zoo_model("keypoint-rcnn-resnet50-fpn-coco-torch")
dataset.apply_model(model, label_field="predictions")

session = fo.launch_app(dataset)
```

Reply:
I actually implemented it earlier and decided to not include it in this PR, because it seemed per-point confidences need further processing (sigmoid?). The output values looked like this.

` 'keypoints_scores': tensor([[16.0222, 18.2254, 17.0186, 11.8371, 13.9270,  6.0463,  7.8868,  7.1934,
          6.9537,  7.8964, 11.9600, -0.3767, -2.9283, -2.6453, -1.6882, -2.4117,
         -3.3753],
        [ 3.4408,  7.4437,  3.9761,  6.6165,  4.9816, -0.2632, -1.3853, -1.6286,
         -1.7510,  1.1719, -1.5498, -1.6644, -1.3459,  0.8909, -2.4248,  0.2460,
         -2.5256],
        [ 1.0068,  1.3324,  1.4261,  3.2719,  2.5410, -0.2752, -0.0260,  1.1362,
          1.7262,  1.0567, -0.1256, -1.9884, -2.3250, -2.7186, -1.9209, -2.9283,
         -2.7841]`
         
         I could either relax the description of confidences to be arbitrary float value (which changes the meaning of the field) and pass this value as is or apply sigmoid-like function to the output. However, since this output isn't really documented on torchvision side, I decided to not populate it for now. (https://pytorch.org/vision/stable/_modules/torchvision/models/detection/keypoint_rcnn.html#keypointrcnn_resnet50_fpn)
    
If you have insight into the expected value of this "keypoint_scores", let me know!

Reply:
I think your intuition apply sigmoid is likely to be correct. How about wiring that up and then checking if `sigmoid(keypoints_scores)` make sense with some of the model outputs?

That'll be easy once the App supports renderings `confidences`, but in the meantime you can probably figure it out by printing some keypoints and eyeballing in the App

Reply:
Added per-point confidences, was debating whether to apply confidence_thresh to each point or leave it to the user since this probably means visible flag needs to be toggled off if it's lower than threshold (the model outputs all 1's for its keypoint prediction). I left it intact for now.

Reply:
I think it might be useful for the app to handle filtering (so user can adjust threshold and see the visualization change dynamically) rather than having a static threshold value.

Reply:
I'm not on board with this approach. If we want App filtering per point, then points should be treated as real entity with `id`s. Parallel lists is not a safe way to store data.

Reply:
I agree that we need to consider filtering support carefully.

One important point here is that a `Keypoint` object definitively represents a semantic grouping of points. In that sense, if we *do* support filtering individual keypoints within an object, it is a different sort of operation than filtering other things like bounding boxes. An example of this is the upcoming [keypoint skeleton](https://github.com/voxel51/fiftyone/issues/1563) that we'll be adding. If a user is allowed to filter points in a skeleton by confidence, the `Keypoint` representation will have to be implemented such that we can always associate individual points with their skeleton vertex. This effectively means each point needs an "index", which suggests a list is indeed a reasonable data structure.

I'm not in favor of assigning `ObjectId`s to individual points within a `Keypoint`, for a number of reasons, including:
- Introduces an additional level of ID nesting not present for other label types
- Sets a precedent that we should add IDs to bounding box coordinates and polyline vertexes
- Would necessitate non-trivial breaking changes to the data model

My current proposal would be that filtering `Keypoint.points` by confidence can be implemented by either:
- Replacing low confidence points by `NaN`, since there is already a precedent that `NaN`-valued points are allowed but cause the point to not be rendered
- Replacing the corresponding `visible[i]` with `False`
 
With respect to breaking data model changes: this project is sufficiently popular at this point(!) that refactoring `Keypoint.points` into something other than a list of lists gives me pause, since it may break existing user code. @stbdang made the shrewd observation that models like `keypoint-rcnn-resnet50-fpn-coco-torch` do not produce any notion of `Keypoint.confidence`, only individual point confidences. So, I'm willing to move from `Keypoint.confidence` to `Keypoint.confidences` under the assumption that there are not likely to be important workflows that rely on filtering by `Keypoint.confidence`

Reply:
Hey @stbdang really appreciate your work on this. There's two remaining blockers here:

1. We need to make an App tweak one way or the other to reflect the fact that `confidence` is no longer a field on `Keypoint` classes. Specifically, we either need to remove the `confidence` filter altogether for `Keypoint`-type fields, or we need to keep the filter but update the filtering logic per my comments above so that it will insert `NaN` values to achieve the desired effect of "hiding" those keypoints
2. I'm on the fence on whether to introduce the convention of populating a `visible` field for keypoints right now, because this isn't implemented by other label types like `Detection` and `Polyline`